src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -2708,14 +2667,54 @@ class DrawSketchHandlerDimension : public DrawSketchHandler
         return false;
     }
 
+    void finishDimensionCreation(int GeoId1, int GeoId2, Base::Vector2d onSketchPos)
+    {
+        bool fixed = GeoId2 == GeoEnum::GeoUndef ? isPointOrSegmentFixed(Obj, GeoId1) : areBothPointsOrSegmentsFixed(Obj, GeoId1, GeoId2);
+
+        int index = Obj->Constraints.getValues().size() - 1;
+        if (fixed || constraintCreationMode == Reference) {
+            Gui::cmdAppObjectArgs(Obj, "setDriving(%i,%s)", index, "False");
+        }
+
+        addConstraintIndex();
+        moveConstraint(index, onSketchPos);
+    }
+
+    void addConstraintIndex()
+    {
+        cstrIndexes.push_back(Obj->Constraints.getValues().size() - 1);
+    }
+
+    bool hasBeenAborted()
+    {
+        // User can abort the command with Undo (ctrl-Z)
+        if (cstrIndexes.size() > 0) {
+            if (cstrIndexes[cstrIndexes.size() - 1] != Obj->Constraints.getValues().size() - 1 ) {
```

Comment:
Consider using `.back()`, e.g.
```suggestion
            if (cstrIndexes.back() != Obj->Constraints.getValues().size() - 1 ) {
```

Reply:
Oh yes back() is good.